File: frontend/src/api/compliance.ts
import { apiClient } from "./client";

export interface ComplianceRequest {
  language: string;
  message: string;
}

export interface ComplianceResult {
  compliant: boolean;
  score: number;
  issues: string[];
  suggestions: string[];
  message: string;
}

export async function checkCompliance(
  data: ComplianceRequest
): Promise<ComplianceResult> {
  const response = await apiClient.post<ComplianceResult>(
    "/compliance/check",
    data
  );

  return response.data;
}

File: backend/app/schemas/compliance.py
from pydantic import BaseModel, Field


class ComplianceRequest(BaseModel):
    message: str = Field(..., min_length=1)
    language: str = "en"


class ComplianceResponse(BaseModel):
    compliant: bool
    score: int
    issues: list[str]
    suggestions: list[str]
    message: str